site/instrument.ts

```diff
@@ -34,4 +35,39 @@ if (LOAD_SENTRY) {
         release: COMMIT_SHA,
         ...sentryOpts,
     })
+
+    // Set Google Analytics client ID as Sentry user ID for session replays
+    if (analyticsConsent && !isInIFrame()) {
```

Comment:
In the description, you say:

> This is achieved using a robust polling mechanism to wait for the GA cookie, accommodating users who accept cookies later in their session.

Won't `analyticsConsent` be false if the user didn't accept the analytics yet? Unless I'm missing something, we'd only start the polling after the consent was already given, in which case we might not need to poll at all.

If I'm wrong about this, please add a comment to the code that explains it.